### PR TITLE
Openstack support for non loadbalancer deployments

### PR DESCRIPTION
**Pull request by drekle**
resolves: 
https://github.com/kubernetes/kops/issues/6593
https://github.com/kubernetes/kops/issues/6592

The loadbalancer can be disabled by not specifiying an api loadbalancer type.
Non loadbalancer deployments will generate a certificate with the 3 master floating IP addresses.
Kubeconfig will choose arbitrarily between each master.

/sig openstack

### CODE REVIEW

File: pkg/model/openstackmodel/firewall.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (b *FirewallModelBuilder) addHTTPSRules(c *fi.ModelBuilderContext, sgMap ma
 	addDirectionalGroupRule(c, masterSG, nodeSG, httpsIngress)
 	addDirectionalGroupRule(c, masterSG, masterSG, httpsIngress)
 
-	if b.UseLoadBalancerForAPI() {
```
Comment (drekle) [inline pkg/model/openstackmodel/firewall.go]: This utility method checks whether `cluster.Spec.API.LoadBalancer` is nil.  I am unsure of the conditions in which this is true, however in create cluster it seems this is always set to an empty struct if the following conditions are true:
```
	return s.DNS == nil && s.LoadBalancer == nil
```
I am using gossip without a loadbalancer in this scenario, connecting to one of the 3 masters.  I would not consider this to be a production ready configuration which makes me question whether these changes should be warranted at all.

**Comment by drekle** [comment]
/hold

I havent tested update after update.

**Comment by zetaab** [comment]
for me it looks okay. However, I must test this in following scenarios:
1) create new cluster with LB, recreate all machines once and check is cluster health after that
2) create new cluster without LB, recreate all machines once and check is cluster health after that

**Comment by drekle** [comment]
> Ok, I took `--os-lb-floating-subnet` away because we are not using lb here (anyways that could be checked in code).

I've fixed this, yet to push.  I cannot do an update after an update for some reason yet in my testing.  Continuing to hold.
```
error running task "Subnet/utility-<ZONE>.derek.k8s.local" (7m26s remaining to succeed): Field cannot be changed: Network
```

**Comment by drekle** [comment]
/remove hold

@zetaab Please take a look at your convenience.  This was more complex than I thought.

- I've moved the create_cluster bit which sets the lb subnet into the check to see if lb's are used.
- I've cleaned up the Subnet code which was previously checking if a pointer was set on expected, instead of comparing the changes struct or comparing the difference between actual and expected.
- I've added some methods so that the masterkeypair task can find a floating IP from the floatingiptask which uses a server instead of a loadbalancer.

**Comment by drekle** [comment]
closing in favor of https://github.com/kubernetes/kops/pull/6608 which is cleaned up and does not have this gofmt issue

**Pull request closed without merging**